src/components/container/OverviewTable/OverviewTable.tsx

```diff
@@ -0,0 +1,146 @@
+import React, { CSSProperties, useContext, useState } from 'react';
+import { ColorDefinition, computeThresholdDays, getDailyDataTypeDefinition, getSurveyDataProvider, isSurveyDataType, NotEnteredThreshold, OverviewData, OverviewDataProvider, OverviewDataType, PrimaryOverviewDataType, resolveColor, SecondaryOverviewDataType, useInitializeView } from '../../../helpers';
+import { LayoutContext, LoadingIndicator } from '../../presentational';
+import './OverviewTable.css';
+import { add, startOfToday } from 'date-fns';
+import { createPreviewDataProvider } from './OverviewTable.previewData';
+import { IconDefinition } from '@fortawesome/fontawesome-svg-core';
+import { FontAwesomeSvgIcon } from 'react-fontawesome-svg-icon';
+import language from '../../../helpers/language';
+
+export interface OverviewTableProps {
+    preview?: boolean;
+    primaryDataType: PrimaryOverviewDataType;
+    secondaryDataTypes: SecondaryOverviewDataType[];
```

Comment:
Usage-wise, I'll admit that I just spent a surprisingly long time understanding the fact that data *only shows up in the table* if you pass it in as a Secondary type.  This is... kind of confusing?  I was lost for a little bit that setting the Primary type to "Mood" doesn't cause the mood *values* to show up in the table.  I think these two inputs could use some naming thought, because Primary and Secondary don't really convey their usage (to me, at least).  It's also a little confusing for me because you're not actually defining a "data type" (there's a data type *in* this type, accompanied by customization parameters).

Alternatively, maybe it's just the setup of the story that lead me down the confusing path, as the default state passes *all* of the types as secondary data types.  Because of this, the primary is set to Mood, but Mood is also the leftmost column of the table, so it made me think that was... purposeful?  

I could be way off base here, this was just my impression coming in without a frame of reference.  I think I'd go with something like `OverviewTableGroupByDataParameters` and `OverviewTableComparisonDataParameters`.

Reply:
I like that suggestion.  The "type in a type" thing did feel a tad off to me.

How about `OverviewTableGroupByDataConfiguration` and `OverviewTableComparisonDataConfiguration`?  "Parameters" doesn't feel like quite the right word to me, but I don't feel that strongly about it either way.

RE: the story setup - I went back and forth a couple of times on that, but I wanted to make all of the data types available to be either the primary or a secondary, for demonstration purposes.  I could adjust the default to not include mood rating in the comparison data types, if you think that would help.

Reply:
I have renamed these to `InsightMatrixGroupByDataConfiguration` and `InsightMatrixComparisonDataConfiguration`.

I have also removed "mood" from the default comparison types on the storybook.